### DISCUSSION TITLE
bug: @trpc/server needed in the frontend for types to work

### DISCUSSION THREAD

**Discussion by Lilja** [trpc/trpc#2973] [Q&A] [answered]
### Provide environment information

```  
System:
    OS: macOS 12.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 131.52 MB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 18.8.0 - ~/.local/share/nvm/v18.8.0/bin/node
    Yarn: 1.22.19 - ~/.local/share/nvm/v18.8.0/bin/yarn
    npm: 8.18.0 - ~/.local/share/nvm/v18.8.0/bin/npm
  Browsers:
    Chrome: 106.0.5249.103
    Firefox: 105.0.2
    Safari: 15.3
  npmPackages:
    @trpc/client: ^10.0.0-proxy-beta.17 => 10.0.0-proxy-beta.17
```

### Describe the bug

Typescript can't complete the typing of a query, see discussion https://github.com/trpc/trpc/discussions/2963

![image](https://user-images.githubusercontent.com/6134511/195023305-ec1e0e82-1b7b-45d0-b8c2-d9cddd564ba7.png)
`Property 'subscribe' does not exist on type 'DecoratedProcedureRecord<unknown, BuildProcedure<"subscription", { _config: RootConfig<{ ctx: {}; meta: {}; errorShape: never; transformer: CombinedDataTransformer; }>; _ctx_out: {}; _input_in: unique symbol; _input_out: unique symbol; _output_in: u
 nique symbol; _output_out: unique symbol; _meta: {}; }, Observable<....'.`

### Link to reproduction

https://github.com/Lilja/trpc-v10-server-needed

### To reproduce

Open up `src/index.ts` and modify the `type {AppRouter}` import, point it to the trpc repo OR a another router.

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

**Comment by KATT** [comment, marked as answer]
Aren't you getting a peer dependency error here? You need to install `@trpc/server`

**Reply by MaxVhanamane** [reply]
Why do you need @trpc/server on the frontend?
I initially attempted to run the development server without installing '@trpc/server' on the frontend, which resulted in this error occurring whenever I ran the server.

[ERROR] Could not resolve "@trpc/server/observable"

    node_modules/@trpc/client/dist/links/httpLink.mjs:1:27:
      1 │ import { observable } from '@trpc/server/observable';
      
 After reading your comment I installed @trpc/server on the frontend now everything is working fine. but my question is why do you need @trpc/server package on the frontend.

**Reply by KATT** [reply]
> Why do you need @trpc/server on the frontend?

The alternative would be to have a third package called `@trpc/core` or something like that that both server and client needed to require

**Reply by MaxVhanamane** [reply]
I was concerned about including '@trpc/server' on the client side because it increases the bundle size. I looked for the '@trpc/core' package, but there isn't one available on npm. If it's necessary to use both '@trpc/client' and '@trpc/server' on the client side, I'm fine with that.
